app/connectors/JourneyConnector.scala

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import common.TaxYear
+import uk.gov.hmrc.http.HeaderCarrier
+
+trait JourneyConnector[F[_], M, R] {
```

Comment:
I wouldn't generalize it just yet. 

Usually [F, E, A] means F is an effect, E is an error, and A is a success. For our use case Future cannot be replaced easily as we discussed the other day so I wouldn't pretend it would be possible in future.

I wouldn't try to have common trait unless you're going to use StubImpl in your tests